specification/http/1.0/openapi.json

```diff
@@ -0,0 +1,477 @@
+          "MyToolkit.MyTool",
+          "MyToolkit.MyTool@1.0",
+          "Calculator.Add@latest"
+        ]
+      },
+      "ToolSecret": {
+        "type": "object",
+        "properties": {
+          "id": {
+            "type": "string",
+            "description": "The secret's unique identifier."
+          },
+          "value": {
+            "type": "string",
+            "description": "The secret's value."
+          }
+        },
+        "required": ["id", "value"],
+        "additionalProperties": false
+      },
+      "SimpleJSONSchema": {
+        "type": "object",
+        "additionalProperties": true,
+        "not": {
+          "properties": {
+            "$ref": {},
+            "$defs": {},
+            "definitions": {}
+          }
+        }
+      },
+      "ToolDefinition": {
+        "type": "object",
+        "properties": {
+          "id": {
+            "$ref": "#/components/schemas/ToolId"
+          },
+          "name": {
+            "type": "string",
+            "description": "The tool's name."
+          },
+          "description": {
+            "type": "string",
+            "description": "A human-readable explanation of the tool's purpose. This field can be used by both humans and AI models."
+          },
+          "toolkit": {
+            "type": "object",
+            "properties": {
+              "name": {
+                "type": "string",
+                "description": "The name of the toolkit."
+              },
+              "description": {
+                "type": "string",
+                "description": "A human-readable description of the toolkit."
+              },
+              "version": {
+                "type": "string",
+                "description": "An identifier for this version of the toolkit."
+              }
+            },
+            "required": ["name", "description"],
+            "additionalProperties": false
+          },
+          "input_schema": {
+            "type": "object",
+            "properties": {
+              "parameters": {
+                "$ref": "#/components/schemas/SimpleJSONSchema",
+                "description": "JSON Schema describing the input parameters for the tool. Supports standard JSON Schema validation but excludes $ref and definitions/schemas for simplicity.",
+                "additionalProperties": true,
+                "required": ["name", "description"]
+              }
+            },
+            "required": ["parameters"],
+            "additionalProperties": false
+          },
+          "output_schema": {
+            "type": "object",
+            "properties": {
+              "possible_kinds": {
+                "type": "array",
+                "description": "The kinds of output that the tool can return: none (a void- or None-returning function), or json (a function that returns a JSON-like value).",
+                "items": {
+                  "type": "string",
+                  "enum": ["none", "json"]
```

Comment:
@eyurtsev I am curious for your thoughts here. I renamed `value` to `json` to explicitly indicate that the output value is free-form JSON. By not using the name `value` I feel like we keep open the possibility of adding other kinds in the future, stuff like `blob, image, artifact...`

Reply:
Kept the enum values as `none || json` but renamed the output field back to `output.value`.